### PR TITLE
Fix proptype warnings, add better i18n support

### PR DESCRIPTION
**Pull request by job13er**
### This project uses [semver](semver.org), please check the scope of this pr:

- [ ] #none# - documentation fixes and/or test additions
- [ ] #patch# - backwards-compatible bug fix
- [x] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG
 * **Fixed** prop type warning caused by not setting `hook` on `frost-scroll`
 * **Enabled** prop types to throw errors instead of warnings, so errors like the above don't happen again
 * **Added** an optional `refineByLabel` to `frost-object-browser` to better support `i18n`, allowing consumers to pass in a translated label (nested under an `i18n` property).
 * **Added** an optional `getSelectedItemsLabel` callback to `frost-action-bar` to better support `i18n`, allowing consumers to provide a callback that will generate a translated label based on the number of selected items (nested under an `i18n` property).
 * **Deprecated** the `string-pluralize` helper as it's no longer needed by this addon and should be removed in the next major version.

### CODE REVIEW

File: addon/components/frost-action-bar.js [hunk 1]
```diff
@@ -26,7 +26,10 @@ export default Component.extend({
     selectedItems: PropTypes.arrayOf(PropTypes.oneOfType([
       PropTypes.EmberObject,
       PropTypes.object
-    ])).isRequired
+    ])).isRequired,
+
+    // callbacks
+    getSelectedItemsLabel: PropTypes.func
```
Comment (sglanzer) [inline addon/components/frost-action-bar.js]: What do you think of creating a special prefix for anything intended strictly for i18n?  E.g.

```
i18n: PropTypes.shape({
  selectedItems: PropTypes.func
})
```

Then when I'm looking at the interface I know exactly what it's intended for and the "translation" portion of the interface is quite clear

**Comment by sglanzer** [comment]
Approved - no issues with this, but I definitely would like to promote a convention for how we identify/group `i18n` properties in interfaces

File: addon/components/frost-action-bar.js (hunk 1 above)
Comment (job13er) [inline addon/components/frost-action-bar.js]: Sure.

Comment (job13er) [inline addon/components/frost-action-bar.js]: Done

File: package.json [hunk 2]
```diff
@@ -106,6 +105,6 @@
     "configPath": "tests/dummy/config"
   },
   "pr-bumper": {
-    "coverage": 82.26
+    "coverage": 76.34
```
Comment (job13er) [inline package.json]: The coverage numbers being reported were actually not accurate. I upgraded `ember-cli-code-coverage` and pinned `ember-code-snippets` to allow a more accurate number and reset the baseline here.

**Comment by sglanzer** [comment]
Approved - nice!

**Pull request merged**